platforms/ios/lib/WysiwygComposer/Sources/HTMLParser/Extensions/DTCoreText/DTHTMLElement.swift

```diff
@@ -21,6 +21,16 @@ extension DTHTMLElement {
     func sanitize() {
         guard let childNodes = childNodes as? [DTHTMLElement] else { return }
 
+        if tag == .a,
+           attributes["data-mention-type"] != nil,
+           let textNode = self.childNodes.first as? DTTextHTMLElement {
+            let mentionTextNode = MentionTextNodeHTMLElement(from: textNode)
+            removeAllChildNodes()
+            addChildNode(mentionTextNode)
+            mentionTextNode.inheritAttributes(from: self)
+            mentionTextNode.interpretAttributes()
+        }
```

Comment:
Having read a bit about it yesterday and today, I think the custom tag could be a really good way to go. I think it would be good to discuss what this solution would look like before starting work on it. For example would it be split out into `<mention-user>`, `<mention-room>` etc? I think to be valid custom html they have to be hyphenated, but I think this would be a change that helps neaten up some client side code.

Reply:
I think it makes sense too as we're coding the type into the tag, which seems neater to me than having an attribute that needs parsing (and so also needs stripping)